### PR TITLE
docs(ui-shell): add JSDoc annotations

### LINKED ISSUE #5668: UI-shell Header - Vanilla JS
## What package(s) are you using?
- [X] `carbon-components`

## Summary

Trying to implement the vanilla js version of ui-shell header but i'm unable to really find any example to get going. Can someone point me in the right direction? All of the examples [here](https://www.carbondesignsystem.com/components/UI-shell-header/code) use React.

Experimental flag for 'ui-shell' have been set to true and I've grabbed html from the only reference I can find, located [here](https://the-carbon-components.netlify.com/?nav=ui-shell), but shouldn't there be some JS initialization somewhere?

Thanks,
Robert

### PR DESCRIPTION
**Pull request by asudoh**
Closes #5668.

#### Changelog

**New**

- JSDoc annotations for vanilla UI shell code.

### CODE REVIEW

File: packages/components/src/components/ui-shell/side-nav.js [hunk 1]
```diff
@@ -118,6 +134,16 @@ export default class SideNav extends mixin(
    * @member SideNav.options
    * @type {object}
    * @property {string} selectorInit The data attribute to find side navs.
+   * @property {string} [selectorSideNavToggle] The CSS selector to find the toggle button.
+   * @property {string} [selectorSideNavSubmenu] The CSS selector to find the trigger buttons for sub nav items.
+   * @property {string} [selectorSideNavItem] The CSS selector to find the nav items.
+   * @property {string} [selectorSideNavLink] The CSS selector to find the interactive potions in non-nested nav items.
```
Comment (emyarod) [inline packages/components/src/components/ui-shell/side-nav.js]:
```suggestion
   * @property {string} [selectorSideNavLink] The CSS selector to find the interactive portions in non-nested nav items.
```

File: packages/components/src/components/ui-shell/header-submenu.js [hunk 2]
```diff
@@ -16,11 +16,22 @@ import eventMatches from '../../globals/js/misc/event-matches';
...
   handles
 ) {
+  /**
+   * Sub menus in header nav.
+   * @extends CreateComponent
+   * @extends InitComponentBySearch
+   * @extends Handles
+   * @param {HTMLElement} element The element working as a submenu in header nav.
+   * @param {object} [options] The component options.
+   * @param {string} [options.selectorTrigger] The CSS selector to find the trigger butotn.
```
Comment (emyarod) [inline packages/components/src/components/ui-shell/header-submenu.js]:
```suggestion
   * @param {string} [options.selectorTrigger] The CSS selector to find the trigger button.
```

File: packages/components/src/components/ui-shell/header-submenu.js [hunk 3]
```diff
@@ -282,6 +293,9 @@ export default class HeaderSubmenu extends mixin(
    * @member HeaderSubmenu.options
    * @type {object}
    * @property {string} selectorInit The data attribute to find side navs.
+   * @property {string} [selectorTrigger] The CSS selector to find the trigger butotn.
```
Comment (emyarod) [inline packages/components/src/components/ui-shell/header-submenu.js]:
```suggestion
   * @property {string} [selectorTrigger] The CSS selector to find the trigger button.
```

File: packages/components/src/components/ui-shell/side-nav.js [hunk 4]
```diff
@@ -118,6 +134,16 @@ export default class SideNav extends mixin(
...
+   * @property {string} [selectorSideNavLinkCurrent]
+   *   The CSS selector to find the interactive potion in active non-nested nav item.
+   * @property {string} [classSideNavExpanded] The CSS class for the expanded state.
+   * @property {string} [classSideNavItemActive] The CSS class for the active/inactive state for nav items.
+   * @property {string} [classSideNavLinkCurrent]
+   *   The CSS class for the active/inactive state of the interactive potion in non-nested nav items.
```
Comment (emyarod) [inline packages/components/src/components/ui-shell/side-nav.js]:
```suggestion
   *   The CSS class for the active/inactive state of the interactive portion in non-nested nav items.
```

**Review by emyarod — APPROVED** [review]
looks good to me, just a few small changes

**Review by joshblack — APPROVED** [review]

**Review by tw15egan — APPROVED** [review]
👍 ✅

**Review by aledavila — APPROVED** [review]

**Comment by asudoh** [comment]
Thanks @emyarod for your feedback and @tw15egan for applying the correction!

**Pull request merged**